internal/db/schema/migrations/oss/postgres/99/04_views.up.sql

```diff
@@ -0,0 +1,73 @@
+begin;
+
+-- worker_aggregate view allows the worker and configuration to be read at the
+-- same time.
+create view server_worker_aggregate as
+with worker_config_tags(worker_id, source, tags) as (
+  select
+    ct.worker_id,
+    ct.source,
+    -- keys and tags can be any lowercase printable character so use uppercase characters as delimitors.
+    string_agg(distinct concat_ws('Y', ct.key, ct.value), 'Z') as tags
+  from server_worker_tag ct
+  group by ct.worker_id, ct.source
+),
+ session_count (worker_id, count) as (
```

Comment:
A session can only be terminated if all its connections are closed. If we added back the `worker_id` column to the session table, this could be a more straightforward query based on whether or not `termination_reason` has been set for the session.

Reply:
Looks good to me!